### PR TITLE
Allow adminstrator teams members to see other teams

### LINKED ISSUE #18885: Unable to see group members
### Gitea Version

1.16.2

### Git Version

N/A

### Operating System

N/A

### How are you running Gitea?

N/A

### Database

_No response_

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Description

If user contains group with administration access, user can see settings in the repo, but can not see groups which have access to that repo. 

### Screenshots

image
image

### PR DESCRIPTION
**Pull request by Gusted**
- Allow team members with adminstrator access, to view other teams.
- Resolves https://github.com/go-gitea/gitea/issues/18885

### CODE REVIEW

**Review by silverwind — APPROVED** [review]

**Comment by wxiaoguang** [comment]
It's better to change the concept from `admin` to `manager`, then no confusion anymore.

`admin` sounds like it can control everything, while `manager` sounds like it can only manage what it can manage.

**Comment by Gusted** [comment]
> It's better to change the concept from `admin` to `team manager`, then no confusion anymore.
> 
> `admin` sounds like it can control everything, while `manager` sounds like it can only manage what it can manage.

Are you quitly trying to insert a proposal here 😄?

**Comment by wxiaoguang** [comment]
> > It's better to change the concept from `admin` to `team manager`, then no confusion anymore.
> > `admin` sounds like it can control everything, while `manager` sounds like it can only manage what it can manage.
> 
> Are you quitly trying to insert a proposal here 😄?

Hmm, yes .... just a personal opinion to make the concept clear,  please ignore me ~~~

**Review by zeripath — APPROVED** [review]

**Comment by zeripath** [comment]
Blocked As per discussion in #maintainers 

@lunny - do you think we should merge as is as is backportable to 1.16 and then move to a separate unit to 1.17

File: modules/context/org.go [hunk 1]
```diff
@@ -129,7 +129,11 @@ func HandleOrgAssignment(ctx *Context, args ...bool) {
 
 	// Team.
 	if ctx.Org.IsMember {
-		if ctx.Org.IsOwner {
+		maxPerm, err := org.GetOrgUserMaxAuthorizeLevel(ctx.User.ID)
+		if err != nil {
+			ctx.ServerError("GetOrgUserMaxAuthorizeLevel", err)
```
Comment (lunny) [inline modules/context/org.go]: need return

File: modules/context/org.go [hunk 2]
```diff
@@ -129,7 +129,11 @@ func HandleOrgAssignment(ctx *Context, args ...bool) {
...
+		}
+		if ctx.Org.IsOwner || maxPerm >= perm.AccessModeAdmin {
```
Comment (lunny) [inline modules/context/org.go]: I think we should check if the user is in a team with admin permission and include all repositories in this organization.

**Review by 6543 — APPROVED** [review]

File: modules/context/org.go [hunk 3]
```diff
@@ -129,7 +129,23 @@ func HandleOrgAssignment(ctx *Context, args ...bool) {
 
 	// Team.
 	if ctx.Org.IsMember {
+		shouldSeeAllTeams := false
 		if ctx.Org.IsOwner {
+			shouldSeeAllTeams = true
+		} else {
+			teams, err := org.GetUserTeams(ctx.User.ID)
+			if err != nil {
+				ctx.ServerError("GetOrgUserMaxAuthorizeLevel", err)
```
Comment (wxiaoguang) [inline modules/context/org.go]:
```suggestion
				ctx.ServerError("GetUserTeams", err)
```

**Comment by 6543** [comment]
:rocket:

**Pull request merged**